File: EndeavourModels/BaseTable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EndeavourModels
{
    public class BaseTable
    {
        public DateTime InsertedDate { get; set; }
        public string InsertedBy { get; set; } = "";
        public DateTime UpdatedDate { get; set; }
        public string UpdatedBy { get; set; } = "";
        public bool IsActive { get; set; }
    }
}

File: EndeavourDrinks.ng/src/app/models/baseTable.ts
export interface IBaseTable {
  insertedDate: Date;
  insertedBy: string;
  updatedDate: Date;
  updatedBy: string;
  isActive: boolean;
}

export class BaseTable implements IBaseTable {
  public insertedDate: Date = new Date();
  public insertedBy: string = '';
  public updatedDate: Date = new Date();
  public updatedBy: string = '';
  public isActive: boolean = false;

  constructor(init?: Partial<IBaseTable>) {
    Object.assign(this, init);
  }
}
